src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -234,7 +234,24 @@ private void ValidateLinuxSKUConfiguration()
                 _environment.IsLinuxConsumptionOnAtlas() &&
                 !_environment.IsManagedAppEnvironment())
             {
-                _logger.LogError($"Unable to load the functions payload since the app was not provisioned with valid {AzureWebJobsSecretStorage} connection string.");
+                logger?.LogError($"Unable to load the functions payload since the app was not provisioned with valid {AzureWebJobsSecretStorage} connection string.");
+            }
+
+            var functionsTimeZone = _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsTimeZone);
+            var functionsWebsiteTimeZone = _environment.GetEnvironmentVariable(EnvironmentSettingNames.FunctionsWebsiteTimeZone);
+
+            // If we have a linux consumption app and the time zone env variable is set, log a warning and diagnostic event
+            if (EnvironmentExtensions.IsLinuxConsumptionOnAtlas(_environment) &&
```

Comment:
nit -- can you switch this around to be `_environment.IsLinuxConsumptionOnAtlas()`?